syntax = "proto3";

service CatalogService {
    rpc GetBookInfo (BookRequest) returns (BookResponse);
    rpc ListBooks (Empty) returns (BookList);
}

message BookRequest {
    string title = 1;
}

message BookResponse {
    int32 id = 1;
    string title = 2;
    string author = 3;
    int32 year = 4;
    int32 quantity = 5;
    double price = 6;
}

message BookList {
    repeated BookResponse books = 1;
}

message Empty {}
